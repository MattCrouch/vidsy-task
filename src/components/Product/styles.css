.product {
  display: inline-block;
  width: 30%;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
  padding: 1em;
  margin: 0 3% 1em 0;
}

.product__thumbnail {
  display: block;
  margin: 0 auto;
  max-width: 100%;
  max-height: 8em;
}

.product__name, .product__brand {
  margin: 0;
}

.product__brand {
  font-weight: normal;
}

.product__color, .product__size {
  display: inline-block;
  color: darkgray;
  margin-right: 0.5em;
}

.product__price {
  display: block;
}