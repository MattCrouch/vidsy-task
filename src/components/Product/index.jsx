import React from 'react';
import PropTypes from 'prop-types';

import './styles.css';

function Product(props) {
  return (
    <li className="product">
      <img className="product__thumbnail" src={props.thumbnailUri} alt={props.name} />

      <h3 className="product__name">{props.name}</h3>
      <h4 className="product__brand">{props.brand}</h4>
      <span className="product__color">{props.color}</span>
      <span className="product__size">{props.size}</span>
      <span className="product__price">&pound;{props.price / 100}</span>      
    </li>
  );
}


Product.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  brand: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
  size: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  thumbnailUri: PropTypes.string.isRequired,
}


export default Product;