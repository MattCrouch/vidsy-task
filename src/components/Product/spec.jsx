import React from 'react';
import renderer from 'react-test-renderer';
import Product from '../Product';

it('renders correctly', () => {
  const tree = renderer.create(<Product
    id={1}
    name="T-Shirt"
    brand="Brand A"
    color="black"
    size="medium"
    price={123}
    thumbnailUri="images/t-shirt.jpg"
  />).toJSON();
  expect(tree).toMatchSnapshot();
});