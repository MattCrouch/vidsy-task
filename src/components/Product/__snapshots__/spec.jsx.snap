// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li
  className="product"
>
  <img
    alt="T-Shirt"
    className="product__thumbnail"
    src="images/t-shirt.jpg"
  />
  <h3
    className="product__name"
  >
    T-Shirt
  </h3>
  <h4
    className="product__brand"
  >
    Brand A
  </h4>
  <span
    className="product__color"
  >
    black
  </span>
  <span
    className="product__size"
  >
    medium
  </span>
  <span
    className="product__price"
  >
    Â£
    1.23
  </span>
</li>
`;
